app/controllers/application_controller.rb

```diff
@@ -44,7 +44,9 @@ def create_user_event(event_type, user = current_user)
   end
 
   def decorated_session
-    @_decorated_session ||= DecoratedSession.new(sp: current_sp, view_context: view_context).call
+    @_decorated_session ||= DecoratedSession.new(
+      sp: current_sp, view_context: view_context, session: sp_session
```

Comment:
it seems a little confusing to call this named param `session` when we are passing just the `sp_session` -- thoughts on re-naming?